utils.go

```diff
@@ -300,9 +301,13 @@ func (s *Service) detectLocation(name string) (location string, err error) {
 		return
 	}
 
-	// Example URL: https://bucket.zone.qingstor.com
-	location = strings.Split(r.Header.Get(headers.Location), ".")[1]
-	return
+	// Example URL: https://zone.qingstor.com/bucket
+	base , err := url2.Parse(r.Header.Get(headers.Location))
```

Comment:
Refactor

```go
url := fmt.Sprintf("%s://%s:%d/%s", s.config.Protocol, s.config.Host, s.config.Port, name)
```

to

```go
u := fmt.Sprintf("%s://%s:%d/%s", s.config.Protocol, s.config.Host, s.config.Port, name)
```

So that we don't need to alias `net/url` to `url2`.

Reply:
Also, please format code.

Reply:
`base` is not a good name for me, how about `locationUrl`?

Reply:
yes